analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EqualityOnFloatingPoint.CSharp11.cs

```diff
@@ -4,6 +4,24 @@
 
 public class EqualityOnFloatingPoint
 {
+    void TestHalfNaN()
+    {
+        bool b;
+        var h = Half.NaN;
+
+        b = h == Half.NaN; // Noncompliant {{Do not check floating point equality with exact values, use 'Half.IsNaN()' instead.}}
```

Comment:
Add a test case with
```cs
using static System.Double;
_ = d == NaN;
```

You can add this test case in another file as well. I think it was introduced in C#6.

Reply:
Add a test case where `T.IsNan()` is proposed:
```cs
public class C
{
    public void M<T>(T t) where T : IFloatingPointIeee754<T>
    {
        if (t == T.NaN) { } // Uncompliant
        if (T.IsNaN(t)) { } // Compliant
    }
}
```

Reply:
`using static System.Double` is a good one: thanks for spotting it out!

Although probably uncommon, I am going to try to have it work.

Reply:
Easier than expected, it's fixed now.